opensrp-pnc/src/main/java/org/smartregister/pnc/processor/PncMiniClientProcessorForJava.java

```diff
@@ -110,11 +117,56 @@ public void processEventClient(@NonNull EventClient eventClient, @NonNull List<E
                 processPncVisit(eventClient);
                 CoreLibrary.getInstance().context().getEventClientRepository().markEventAsProcessed(eventClient.getEvent().getFormSubmissionId());
                 break;
+            case PncConstants.EventTypeConstants.DEATH:
+                processDeathEvent(eventClient);
+                processEvent(eventClient.getEvent(), eventClient.getClient(), clientClassification);
+                CoreLibrary.getInstance().context().getEventClientRepository().markEventAsProcessed(eventClient.getEvent().getFormSubmissionId());
+                unsyncEvents.add(event);
+                break;
             default:
                 break;
         }
     }
 
+    private void processDeathEvent(@NonNull EventClient eventClient) {
+        Event event = eventClient.getEvent();
+        String entityId = event.getBaseEntityId();
+
+        HashMap<String, String> keyValues = new HashMap<>();
+        generateKeyValuesFromEvent(event, keyValues, true);
+
+        String encounterDateField = keyValues.get("date_of_death");
+
+        ContentValues values = new ContentValues();
+        values.put(PncConstants.KeyConstants.DOD, encounterDateField);
+        values.put(PncConstants.KeyConstants.DATE_REMOVED, PncUtils.getTodaysDate());
+
+        //Update REGISTER and FTS Tables
+        AllCommonsRepository allCommonsRepository = PncLibrary.getInstance().context().allCommonsRepositoryobjects(PncDbConstants.Table.EC_CLIENT);
+        if (allCommonsRepository != null) {
+            allCommonsRepository.update(PncDbConstants.Table.EC_CLIENT, values, entityId);
+            PncLibrary.getInstance().context().allCommonsRepositoryobjects(PncDbConstants.Table.EC_CLIENT).updateSearch(Arrays.asList(new String[]{entityId}));
+        }
+
+        // update date of removal
+        ContentValues contentValues = new ContentValues();
+
+        String encounterDateTimeString = null;
+        if (StringUtils.isNotBlank(encounterDateField)) {
+            try {
+                encounterDateTimeString = formatDate(encounterDateField);
+            } catch (ParseException e) {
+                e.printStackTrace();
+            }
+        }
+
+        if (encounterDateTimeString != null) {
+            contentValues.put(PncConstants.KeyConstants.DATE_REMOVED, encounterDateTimeString);
+        }
+
+        PncLibrary.getInstance().context().getEventClientRepository().getWritableDatabase().update(PncDbConstants.Table.EC_CLIENT, contentValues, PncConstants.KeyConstants.BASE_ENTITY_ID + " = ?", new String[]{entityId});
```

Comment:
you can remove this its done in line 147

Reply:
done